addons/hr_expense/models/account_move.py

```diff
@@ -12,3 +12,8 @@ def button_cancel(self):
             if l.expense_id:
                 l.expense_id.refuse_expense(reason=_("Payment Cancelled"))
         return super().button_cancel()
+
+    def button_draft(self):
+        for line in self.line_ids:
+            line.expense_id.sheet_id.write({'state': 'post'})
```

Comment:
I would add a condition to check that the line has an `expense_id`, like above in `button_cancel()`